core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -609,112 +611,165 @@ public PickResult pickSubchannel(PickSubchannelArgs args) {
     }
   }
 
-  /**
-   * Index as in 'i', the pointer to an entry. Not a "search index."
-   * All updates should be done in a synchronization context.
-   */
   @VisibleForTesting
   static final class Index {
-    private List<EquivalentAddressGroup> addressGroups;
-    private int size;
-    private int groupIndex;
-    private int addressIndex;
+    private List<UnwrappedEag> orderedAddresses = new ArrayList<>();
+    private int activeElement = 0;
+    private boolean enableHappyEyeballs;
 
-    public Index(List<EquivalentAddressGroup> groups) {
+    Index(List<EquivalentAddressGroup> groups, boolean enableHappyEyeballs) {
+      this.enableHappyEyeballs = enableHappyEyeballs;
       updateGroups(groups);
     }
 
-    public boolean isValid() {
```

Comment:
Why was the `public` removed and the methods reordered? It makes the diff harder to read for not difference in behavior. Can you restore the original method order so I don't have to reverse engineer this?

Reply:
Since Index itself is package-private it didn't make sense to me to have methods be public, so was reducing access.  I've restored it.